Self-updater to only offer updates only to newer versions

jdevera opened the pull request:
This PR switches the version comparison of the self-updater from a simple inequality check to a full version comparison.

Fixes #158

jdevera committed:
Enhance self-updater logic to only offer updates when the remote version is newer than the current version

bhou-crto commented on internal/updater/self-updater.go:
```diff
@@ -102,8 +103,10 @@ func (u *SelfUpdater) checkSelfUpdate() <-chan bool {
 
-		ch <- u.latestVersion.Version != u.CurrentVersion &&
+		// Only offer update if remote version is newer than current version
+		shouldUpdate := remote.IsVersionSmaller(u.CurrentVersion, u.latestVersion.Version) &&
```
The rational of comparing exact version here is for the situation to rollback to a previous versions. This happens sometimes when a new version of command launcher is released, but some packages are not compatible to it. As a command launcher admin, having a rollback options is always a good thing. 
So a typical rollback procedure is simply make the `self_update_latest_version_url` return an older version.
With this change, the rollback has to go through a new release process with a newer version.

bhou-crto reviewed (commented):
I think this change makes a big change on the rollback procedure, I would like to see more rational about why it is necessary, and what scenario/problem does it address.

jdevera commented:
Interesting! I had not thought about the rollback case, but it's a good one.
I will give the config option a spin and see what it looks like.

jdevera committed:
Add self-update policy configuration

- Introduces `self_update_policy` configuration option to control update behavior.
- Updates self-updater to apply the configured policy: `exact_match` or `only_newer_version`.
- Adds validation for self-update policy in configuration settings.
- Documentats self-update policies and their implications.
- Tests self-update policy functionality and validation.

jdevera commented:
@bhou-crto I've added the extra config, defaulting to exact_match to keep the current behaviour unless configured otherwise.
